### PR TITLE
Hocs 4389 hocs base image

### PR DESCRIPTION
**Pull request by nickmcmahon01**
Creates a basic base image, initial drone config and readme.

### CODE REVIEW

**Comment by LiamMacP** [comment]
If we're doing this - probably want to get Anchore image scanning on this? Would be good if we could orchestrate a way of ensuring this stays upto date.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.15.0
+
+USER root
+
+RUN apk upgrade --no-cache && apk add --no-cache openjdk17-jre
+
+ENV USER user_hocs
+ENV USER_ID 10000
```
Comment (LiamMacP) [inline Dockerfile:8]: Is this the right USER ID? We use various ones - I believe 1000+ is fine for what we need but worth a check.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.15.0
```
Comment (LiamMacP) [inline Dockerfile:1]: Do we want to remove the point to allow patches at any job trigger?

File: .drone.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+kind: pipeline
+type: kubernetes
+name: build
+
+steps:
+  - name: build & push
+    image: plugins/docker
+    settings:
+      registry: quay.io
+      repo: quay.io/ukhomeofficedigital/hocs-base-image
```
Comment (LiamMacP) [inline .drone.yaml:11]: Is the repo set up on Quay?

Comment (nickmcmahon01) [reply .drone.yaml:11]: no, we (you 😁 ) need to set up the robot to have write permission

File: Dockerfile (hunk 2 above)
Comment (nickmcmahon01) [reply Dockerfile:1]: _maybe_, the base image is to ensure as many layers are shared between services as possible, and to ensure that we know exactly what is running, we can control and update this periodically though.

File: Dockerfile (hunk 1 above)
Comment (nickmcmahon01) [reply Dockerfile:8]: 1000+ are likely to exist on the host system, meaning if you can escape the docker sandbox you have permissions on the host too, we ought to change this to chown 10000:10001

https://github.com/hexops/dockerfile#do-not-use-a-uid-below-10000

**Comment by nickmcmahon01** [comment]
> If we're doing this - probably want to get Anchore image scanning on this? Would be good if we could orchestrate a way of ensuring this stays upto date.
> 
> Might also be worth setting up a nightly cron with a slack webhook that tells us about a failed build on issue so we know to update it.

I agree, but that isn't this PR... but what does Anchore give us over the existing Quay scanning for instance?

**Comment by LiamMacP** [comment]
> > If we're doing this - probably want to get Anchore image scanning on this? Would be good if we could orchestrate a way of ensuring this stays upto date.
> > Might also be worth setting up a nightly cron with a slack webhook that tells us about a failed build on issue so we know to update it.
> 
> I agree, but that isn't this PR... but what does Anchore give us over the existing Quay scanning for instance?

We can break the build on new CVE's so we move the burden of scanning before a merge.

**Comment by nickmcmahon01** [comment]
https://github.com/anchore/scan-action

**Review by LiamMacP — APPROVED** [review]
Approved - change looks good - got next steps to take away.

**Pull request merged**